frontend/apps/filemanager/filemanagerconverter.lua

```diff
@@ -0,0 +1,106 @@
+--[[--
+This module is responsible for converting files.
+]]
+
+local ButtonDialogTitle = require("ui/widget/buttondialogtitle")
+local ConfirmBox = require("ui/widget/confirmbox")
+local UIManager = require("ui/uimanager")
+local lfs = require("libs/libkoreader-lfs")
+local logger = require("logger")
+local util = require("util")
+local gettext = require("gettext")
+local T = require("ffi/util").template
+
+local FileConverter = {
+    formats_from = {
+        md = {
+            name = gettext("Markdown"),
+            from = "markdown",
+        },
+    },
+    formats_to = {
+        epub = {
+            to = "epub",
+        },
+        html = {
+            to = "html",
+        },
+        pdf = {
+            to = "pdf",
+        },
+    },
+    --curl --form input_files[]=@README.md --form from=markdown  --form to=pdf http://c.docverter.com/convert
+    docverter_url = "http://c.docverter.com/convert",
+}
+
+--- Converts a markdown fragment to a full HTML document.
+---- @string markdown the markdown fragment
+---- @string title an optional title for the HTML document
+---- @treturn string an HTML document
+function FileConverter:_mdToHtml(markdown, title)
+    local MD = require("frontend/apps/filemanager/lib/md")
+    local md_options = {
+        prependHead = "<!DOCTYPE html>\n<html>\n<head>\n",
+        appendTail = "\n</body>\n</html>",
+        insertHead = string.format("<title>%s</title>\n</head>\n<body>\n", title),
+    }
+    local html, err = MD(markdown, md_options)
+    if err then
+        logger.warn("FileManagerConverter: could not generate HTML", err)
+    end
+    return html
+end
+
+function FileConverter:_mdFileToHtml(file, title)
+logger.dbg("blaasdfasdfasdfsd")
+    local f = io.open(file, "rb")
+    local content = f:read("*all")
+    f:close()
+    local html = FileConverter:_mdToHtml(content, title)
+    return html
+end
+
+function FileConverter:_writeHtmlToFile(html, file)
```

Comment:
I first wrote it with the .html included, while refactoring I realized it could be made more generic and forgot to change the function name. But yeah, for the quickstart guide I'll need the functionality provided by `FileConverter:_mdToHtml()` and `FileConverter:_writeStringToFile()`.